### ISSUE TITLE
Poor choose of temporary filenames

### ISSUE THREAD

**Issue by jkbonfield** [samtools/samtools#529]
When doing samtools sort from stdin, it uses temporary filenames of STDIN.tmp.[0-9]{4}.bam.  This means two samtools sorts running in parallel read and write each others data, giving borked results.

Now while I *could* be manually picking a unique prefix for each invocation of my pipeline, the defaults ought to just work out of the box.  If anything, I should be able to pick a directory for the prefix and let samtools pick the unique filename component itself.

**Comment by jmarshall** [comment]
Duplicate of #523.
